### DISCUSSION TITLE
What will happen if Notification get de-prioritised for Data-only payload.

### DISCUSSION THREAD

**Discussion by chansk131** [invertase/react-native-firebase#6188] [Q&A] [answered]
According to https://rnfirebase.io/messaging/usage#message-handlers, when we use `data-only` notification payload, we must set `priority:high` to make the app call `setBackgroundMessageHandler` wben it is in background/quit so that we can show local notification.

From https://firebase.google.com/docs/cloud-messaging/android/message-priority, the high priority notifications will get deprioritised if there's not enough user interaction. Is there any concerns if the notification get deprioritised to normal? Will the `setBackgroundMessageHandler` get called when the notification get sent while it is being de-prioritised?

**Comment by mikehardy** [comment, marked as answer]
Your app won't wake up if it is deprioritized. The message will just be dropped and gone.
That can happen for many other reasons too, for instance if your phone is just too hot it will be throttled, and iOS will drop the message

Data-only messages should never be used for application requirements that depend on reliable delivery.
